### PR TITLE
Updated concept page with concepts image and description.

### PR DESCRIPTION
**Pull request by klywang**
## Description

Updated the concepts page with the new overview graphic and added more details about how 

## Motivation and Context

Decreases entry barrier for new users.
Partially addresses #120. More figures and descriptions will be added in separate PRs.

## Checklist:

- [x] I am familiar with the [**Contributing Guidelines**](https://github.com/glotzerlab/signac-docs/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [**Contributor Agreement**](https://github.com/glotzerlab/signac-docs/blob/master/ContributorAgreement.md).
- [x] My code follows the [code style guideline](https://github.com/glotzerlab/signac-docs/blob/master/CONTRIBUTING.md#code-style) of this project.

### CODE REVIEW

File: docs/source/intro.rst [hunk 1]
```diff
@@ -16,4 +16,14 @@ This means that all data is uniquely addressable from the associated parameters.
...
+.. figure:: images/signac_data_space.png
+
+    Overview of the **signac** framework.
+    When a project is created, a workspace directory is intialized on disk.
+    Each job is uniquely identified by a set of user-defined state points, which are formatted as dictionaries.
+    The workspace holds a directory for each job, containing JSON files that store the state point and job document.
+    The job directory name is a hash of the state point's contents.
+    Here, the ``init.py`` file initializes an empty project and adds one job with state point ``{"a": 1}``.
+    Next, users define a workflow using a subclass of **signac-flow**'s ``FlowProject``.
+    The workflow shown has three operations (simulate, analyze, visualize) that, when executed, produce two new files ``results.txt`` and ``plot.png`` in the job directory.
```
Comment (klywang) [inline docs/source/intro.rst:29]: May rephrase this to be more general. We might want to talk about flow in the context of how they fit together, rather than as the next step.

File: docs/source/intro.rst [hunk 2]
```diff
@@ -16,4 +16,14 @@ This means that all data is uniquely addressable from the associated parameters.
 With **signac**, we define the processes generating and manipulating a specific data set as a sequence of operations on a job.
 Using this abstraction, **signac** can define workflows on an arbitrary **signac** data space.
 
-.. image:: images/signac_data_space.png
+.. _concepts_overview
+.. figure:: images/signac_data_space.png
+
+    Overview of the **signac** framework.
+    When a project is created, a workspace directory is intialized on disk.
```
Comment (atravitz) [inline docs/source/intro.rst:23]:
```suggestion
    When a project is initialized (such as `signac init` on the command line) a `workspace/` directory is created.
```

File: docs/source/intro.rst [hunk 3]
```diff
@@ -16,4 +16,14 @@ This means that all data is uniquely addressable from the associated parameters.
...
+    Each job is uniquely identified by a set of user-defined state points, which are formatted as dictionaries.
```
Comment (atravitz) [inline docs/source/intro.rst:24]:
```suggestion
    Each job is identified its `state point`,  a _unique_ dictionary that contains relevant parameters describing the job.
```

we should link to the definition of each of these key words

File: docs/source/intro.rst (hunk 1 above)
Comment (cbkerr) [reply docs/source/intro.rst:29]: Great improvement for now

**Review by cbkerr — CHANGES_REQUESTED** [review]
I think it's great, let's just make a version of the image that does not use a project name to be ready for signac 2.0.

**Comment by vyasr** [comment]
@cbkerr @klywang are you interested in getting this wrapped up and merged now that signac 2 is out?

**Review by tcmoore3 — APPROVED** [review]

**Comment by cbkerr** [comment]
This overview image needs to have signac and signac-flow.

We can have a separate one illustrating the workspace.

**Comment by cbkerr** [comment]
Here is the preview of the docs with these changes: https://signac--137.org.readthedocs.build/en/137/intro.html

**Comment by cbkerr** [comment]
I'm good with this overview now. I asked Joseph for a review as a newer member of the project

**Comment by josephburkhart** [comment]
I just made a few changes based on a conversation today with @cbkerr. Please let me know if you disagree with my suggestions.

**Review by josephburkhart — APPROVED** [review]
Looks good to me now - definitely more precise and the figures look great!

**Review by cbkerr — APPROVED** [review]
